pkg/mdns/mdns.go

```diff
@@ -67,6 +67,11 @@ func (m MDNS) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (i
 	msg.SetReply(r)
 	state := request.Request{W: w, Req: r}
 
+	if state.QName()[len(state.QName())-len(m.Domain)-1:] != m.Domain + "." {
```

Comment:
This relies of m.Domain not ending in ".", but I think we should allow the config option to be either:
`clustername.clusterdomain`
or
`clustername.clusterdomain.`

Reply:
That assumption exists in a few other places here too. I'll fix them all in a subsequent commit.